src/frontend/e2e/pages/upload.ts

```diff
@@ -0,0 +1,84 @@
+import { Page } from "@playwright/test";
+import { faker } from "@faker-js/faker";
+import { SampleData } from "../utils/schemas/sampleData";
+import { UploadData } from "../utils/schemas/uploadData";
+import * as path from 'path';
+
+
+export class UploadSample {
+  public static async uploadSequencingFiles(
+    page: Page,
+    uploadData: UploadData,
+  ): Promise<any> {
+    //click upload button
+    await page.locator("a[href$='/upload/step1']").click();
+    await UploadSample.getSampleDataFile(page,uploadData.dataFile);
+    // complete form
+    for(let i = 0; i < uploadData.samples.length; i++){
+      // fill public ID input
+      // TODO modify selector
+      await page.locator(`//div[contains(@class,'MuiTableBody-root')]/div[@role='row'][${i+1}]/descendant::input[@name='publicId']`).type(uploadData.samples[i].publicId);
+       // fill collection date input
+       // TODO modify selector
+      await page.locator(`//div[contains(@class,'MuiTableBody-root')]/div[@role='row'][${i+1}]/descendant::input[@name='collectionDate']`).type(uploadData.samples[i].collectionDate);
+       // fill search for location input from emergent widget
+       // TODO modify selector
+      await page.locator(`//div[contains(@class,'MuiTableBody-root')]/div[@role='row'][${i+1}]/descendant::button[@label='Search For Location']`).click();
+      await page.locator("//div[@role='tooltip']/descendant::input").type(uploadData.samples[i].collectionLocation,{delay:150});
+      await page.keyboard.press("ArrowDown");
+      await page.keyboard.press("Enter");
```

Comment:
are you testing the apply to all functionality here? That might make the test slow if you're applying all on a loop?

Reply:
@luisczi I wasn't aware you removed the original logic to handle this. Check commit f501e3635cd1ba4423147ba6eb30cec04987d208 and see how you can make use of it to address this feedback

Reply:
@charles-testco the original logic was broken, thats the reason why i changed it, @phoenixAja this is needed since the element is an emergent component that needs focus on specific element prior selecting thats the reason i hit the keyboard, but this doesnt slow it down

Reply:
can you add a comment explaining that please? 😄

Reply:
sure, the script cannot insert the text directly to input because is hidden at component initial state, so after you click on it an emergent widget comes in and that reveals the input into the DOM actual state, i can now proceed to locate the input that i need to insert the text since it works with autocomplete it creates a list of elements on the fly so in order to **avoid to create** extra locator i rather press on those keys to select my entry :)

https://user-images.githubusercontent.com/109995236/190459882-89c4c383-9c7d-4938-b59c-cc94c38afa45.mov

Reply:
oh i meant a comment in the code 😄 but thank you!